File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        std::string result{};

        int min_length = 200;
        for(const auto& s : strs) {
            if (s.size() < min_length)
                min_length = s.size();
        }

        int i = 0;
        while (i < min_length) {
            char c = strs[0][i];
            bool is_match = true;
            for (int j = 1; j < strs.size(); j++) {
                if (strs[j][i] != c) {
                    is_match = false;
                    break;
                }
            }

            if (!is_match) break;
            result += c;
            i++;
        }

        return result;
    }
};

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cs
public class Solution {

    public string Encode(IList<string> strs) {
        var result = new StringBuilder();
        foreach (var s in strs) {
            result.Append(s.Length);
            result.Append('#');
            result.Append(s);
        }

        return result.ToString();
    }

    public List<string> Decode(string s) {
        var result = new List<string>();

        int i = 0;
        while (i < s.Length) {
            //i will always be at the first index of the length 
            //and j will always be at the first index of the string
            int j = i;

            // find the # char
            while (s[j] != '#') j++;
            // get the length of the string
            int length = int.Parse(s.Substring(i, j - i));
            // get the string and put to result
            j++;
            result.Add(s.Substring(j, length));
            i = j + length;
        }

        return result;
   }
}
